GUI/QtQuick/QtQuickAdapter.cpp

```diff
@@ -1061,15 +1072,15 @@ void QtQuickAdapter::walletSelected(int index)
          walletInfoReq_[msgId] = walletName;
 
          if (hdWallets_.count(walletId) > 0) {
+            const auto& hdWallet = hdWallets_.at(walletId);
             walletPropertiesModel_->setWalletInfo({
-               QString::fromStdString(hdWallets_.at(walletId).name),
+               QString::fromStdString(hdWallet.name),
                QString::fromStdString(walletId),
-               QString::fromLatin1("1/") + QString::number(hdWallets_.at(walletId).leaves.size()),
-               hdWallets_.at(walletId).encryptionTypes.size() > 0
-                  ? encTypeToString(hdWallets_.at(walletId).encryptionTypes[0]) : tr("Unknown"),
-               hdWallets_.at(walletId).nbAddresses,
-               hdWallets_.at(walletId).isHardware,
-               hdWallets_.at(walletId).watchOnly
+               QString::fromLatin1("1/") + QString::number(hdWallet.leaves.size()),
+               walletType(hdWallet.isHardware, hdWallet.watchOnly),
+               hdWallet.nbAddresses,
+               hdWallet.isHardware,
+               hdWallet.watchOnly
```

Comment:
I guess it's better to pass hdWallet (as a struct) to `walletPropertiesModel_`, walletType can be created in the model itself, to avoid bloating the adapter code.